ReactNativeClient/lib/reducer.js

```diff
@@ -577,8 +577,20 @@ const reducer = (state = defaultState, action) => {
 			break;
 
 		case 'TAG_UPDATE_ONE':
-			newState = updateOneItem(state, action);
-			newState = updateOneItem(newState, action, 'selectedNoteTags');
+			{
+				newState = updateOneItem(state, action);
+				const availableTags = newState['selectedNoteTags'];
+				let tagExists = false;
+
+				for (let i = 0; i < availableTags.length; i++) {
+					if (availableTags[i].id == action.item.id) {
+						tagExists = true;
+					}
+				}
+
+				if (tagExists)
+					newState = updateOneItem(newState, action, 'selectedNoteTags');
+			}
```

Comment:
This logic is already in `updateOneItem()`. Please add one argument `addIfNotExists = true` to that function, and use it to decide whether or not the item should be added to the array.